src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/EditorConfig/Parsing/Sections/SectionMatch.cs

```diff
@@ -0,0 +1,73 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+namespace Microsoft.CodeAnalysis.EditorConfig.Parsing
+{
+    internal enum SectionMatch
+    {
+        /// <summary>
+        /// Most exact section match for a language. Will always match all files for the given language.
+        /// </summary>
+        /// <remarks>
+        /// - for C# this is [*.cs]
+        /// - for Visual Basic it is [*.vb].
+        /// - If both language are specified it is [*.{cs,vb}]
```

Comment:
Let me try and explain it with just C# as the example:

| Section Pattern | match kind | Description |
|:-----------|-----------|-----------|
|`[*.cs]` | ExactLanguageMatch | This is what we would put if we were asked to generate a section header for this language |
|`[*.{cs,csx}]` | ExactLanguageMatchWithOthers |  it includes things other than this language but contains a language definition similar to `ExactLanguageMatch`. Importantly it does not refer to other know languages |
|`[*.{cs,vb}]`| ExactLanguageMatchWithOthers | it includes things other than this language but contains a language definition similar to `ExactLanguageMatch`. It also included references to other known language (in this case vb). We would not consider this an appropriate place to but a C# only (or VB only) option but if it was placed here it would be respected|
|`[*s]`| SupersetFilePatternMatch| A pattern that is smaller than our "ideal" match case for this section `[*.cs]`  meaning it matches but also includes _more_ filetypes that just `*.cs`|
|`[*.Tests.cs]`| FilePatternMatch| A pattern matches but is more specific that our "ideal" match.|

This is all about answering the question: "We need to add a new option to the file for this language where is the most appropriate place to put it?" It possible that this granularity is unnecessary but `AnyLanguageMatch` is better than `SupersetFilePatternMatch` or `FilePatternMatch` and I could see us wanting to generate a new section only if the match was worse that `AnyLanguageMatch`.